src/decoder.ts

```diff
@@ -383,6 +382,72 @@ export class Decoder<A> {
     });
   }
 
+  /**
+   * Decoder for fixed-length arrays, aka Tuples.
+   *
+   * Supports up to 8-tuples.
+   *
+   * Example:
+   * ```
+   * tuple([number(), number(), string()]).run([5, 10, 'px'])
+   * // => {ok: true, result: [5, 10, 'px']}
+   * ```
+   */
+  static tuple<A>(decoder: [Decoder<A>]): Decoder<[A]>;
+  static tuple<A, B>(decoder: [Decoder<A>, Decoder<B>]): Decoder<[A, B]>;
+  static tuple<A, B, C>(decoder: [Decoder<A>, Decoder<B>, Decoder<C>]): Decoder<[A, B, C]>;
+  static tuple<A, B, C, D>(
+    decoder: [Decoder<A>, Decoder<B>, Decoder<C>, Decoder<D>]
+  ): Decoder<[A, B, C, D]>;
+  static tuple<A, B, C, D, E>(
+    decoder: [Decoder<A>, Decoder<B>, Decoder<C>, Decoder<D>, Decoder<E>]
+  ): Decoder<[A, B, C, D, E]>;
+  static tuple<A, B, C, D, E, F>(
+    decoder: [Decoder<A>, Decoder<B>, Decoder<C>, Decoder<D>, Decoder<E>, Decoder<F>]
+  ): Decoder<[A, B, C, D, E, F]>;
+  static tuple<A, B, C, D, E, F, G>(
+    decoder: [Decoder<A>, Decoder<B>, Decoder<C>, Decoder<D>, Decoder<E>, Decoder<F>, Decoder<G>]
+  ): Decoder<[A, B, C, D, E, F, G]>;
+  static tuple<A, B, C, D, E, F, G, H>(
+    decoder: [
+      Decoder<A>,
+      Decoder<B>,
+      Decoder<C>,
+      Decoder<D>,
+      Decoder<E>,
+      Decoder<F>,
+      Decoder<G>,
+      Decoder<H>
+    ]
+  ): Decoder<[A, B, C, D, E, F, G, H]>;
+  static tuple<A>(decoders?: Decoder<A>[]) {
+    return new Decoder((json: unknown) => {
+      if (isJsonArray(json) && decoders) {
+        if (json.length !== decoders.length) {
+          return Result.err({
+            message: `expected an array of length ${decoders.length}, got an array of length ${
+              json.length
+            }`
+          });
+        }
+        const result = [];
+        for (let i: number = 0; i < decoders.length; i++) {
+          const nth = decoders[i].run(json[i]);
+          if (nth.ok) {
+            result[i] = nth.result;
+          } else {
+            return Result.err(prependAt(`[${i}]`, nth.error));
+          }
+        }
+        return Result.ok(result);
+      } else if (isJsonArray(json)) {
```

Comment:
change request: let's get rid of this boy -- I don't think that allowing a `tuple()` decoder with no arguments makes sense. If you want to check for any array use `array()`, and if you want to check specifically for an empty array use `constant([])`.

Reply:
Totally